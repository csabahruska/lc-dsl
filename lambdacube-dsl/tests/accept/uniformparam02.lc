render modelViewProj = let
  emptyFB           = FrameBuffer (DepthImage #1 1000.0)
  rasterCtx         = LineCtx 1.0 LastVertex
  fragmentCtx       = AccumulationContext (DepthOp Less False)
  vertexShader v    = VertexOut v 1.0 () (Flat v)
  vertexStream      = Fetch "stream" Lines (IV4F "position")
  primitiveStream   = Transform vertexShader vertexStream
  fragmentStream    = Rasterize rasterCtx primitiveStream
  fragmentShader v  = FragmentOutRastDepth ()
  frame             = Accumulate fragmentCtx PassAll fragmentShader fragmentStream emptyFB
 in ScreenOut frame

main = render (Uni (IM44F "MVP"))