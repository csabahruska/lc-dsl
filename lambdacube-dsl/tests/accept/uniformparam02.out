Pipeline {textures = [], samplers = [], targets = [RenderTarget {renderTargets = [(Depth,Just (Framebuffer Depth))]}], programs = [Program {programUniforms = fromList [], programStreams = fromList [("v",("position4",V4F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nin vec4 v ;\nflat out vec4 v0 ;\nvoid main() {\nv0 = v;\ngl_Position = v;\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nflat in vec4 v0 ;\nvoid main() {\n}\n"}], slots = [Slot {slotName = "stream4", slotUniforms = fromList [], slotStreams = fromList [("position4",V4F)], slotPrimitive = Lines, slotPrograms = [0]}], commands = [SetRenderTarget 0,ClearRenderTarget [(Depth,VFloat 1000.0)],SetRasterContext (LineCtx 1.0 LastVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [DepthOp Less False]}),SetProgram 0,RenderSlot 0]}